src/main/java/com/teamscale/upload/autodetect_revision/GitChecker.java

```diff
@@ -1,4 +1,4 @@
-package com.cqse.teamscaleupload.autodetect_revision;
+package com.teamscale.upload.autodetect_revision;
 
 public class GitChecker {
```

Comment:
Why is there no common interface of `GitChecker` and `SvnChecker`.Also, would use `Optional<String>` for the `find` method

Reply:
They each only have 1 static public method, so an Interface makes no sense IMO. They're utility classes. Would you prefer i rename them to GitCheckUtils and SvnCheckUtils?

Reply:
I tried refactoring this to use Optional, including the other two checkers. I reverted it because the code in AutoDetectCommitUtils became an unreadable chain of flatMaps or alternatively the "isPresent then return" antipattern. It feels to me like returning null here makes the consuming code much simpler and easier to read, so I kept that